documentation/using-azure-identity.md

```diff
@@ -0,0 +1,259 @@
+    environment variables?**
+
+    - If so, use the `ManagedIdentityCredential`.
+
+    - If this isn't a concern, use the `EnvironmentCredential`.
+
+- **Is the application deployed to a user device or running in the browser?**
+
+  - **Can the user's device display an authentication site in a browser window
+    or web control?**
+
+    - If so, use the `AuthenticationCodeCredential` or
+      `InteractiveBrowserCredential`.
+
+    - If not, use the `DeviceCodeCredential`.
+
+- **Are you developing an application on your local machine?**
+
+  - Use the `DefaultAzureCredential`.
+
+## Credential Types in @azure/identity
+
+### ClientSecretCredential and ClientCertificateCredential
+
+The `ClientSecretCredential` implements the [client credentials
+flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow)
+to enable confidential clients, like web services, to access Azure resources.
+To use this credential, you will need to create a client secret using the
+"Certificates & secrets" page for your app registration.
+
+The `ClientCertificateCredential` implements the same client credentials flow,
+but instead uses a certificate as the means to authenticate the client. You must
+must generate your own PEM-formatted certificate for use in this flow and then
+[register
+it](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-certificate-credentials#register-your-certificate-with-azure-ad)
+in the "Certificates & secrets" page for your app registration. Using a
+certificate to authenticate is recommended as it is generally more secure than
+using a client secret.
+
+> NOTE: At this time, @azure/identity only supports PEM certificates that are
+> **not** password protected.
+
+For both of these credentials, `tenantId` and `clientId` are required parameters.
+The `clientSecret` or `certificatePath` parameters are also required depending
+on which credential you are using.
+
+### UsernamePasswordCredential
+
+The `UsernamePasswordCredential` follows the [resource owner password credential
+flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth-ropc)
+to authenticate public or confidential clients.  To use this credential, you
+will need the `tenantId` and `clientId` of your app and a `username` and
+`password` of the user you are authenticating.
+
+This credential type supports multi-tenant app registrations so you may pass
+`organizations` as the `tenantId` to enable users from any organizational tenant
+to authenticate.
+
+Generally speaking, we *do not* recommend using this credential type when other
+more secure credential types are available.  Handling the user's password
+directly is a major security risk.
+
+> NOTE: This credential type does not work with personal Microsoft accounts at
+> this time.  See the
+> [documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth-ropc)
+> for more information.
```

Comment:
Might consider adding a comment that this also fails when the user account uses MFA (since the second factor can not be provided).